### ISSUE TITLE
linter: `no-unused-vars` incorrectly reports implicit exports in TypeScript declaration

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26563]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
  "plugins": ["typescript"],
  "rules": {
    "typescript/no-unused-vars": "error"
  }
}
```

### What happened?

```ts
export {};

declare module 'some-package' {
  interface ImplicitlyExportedInterface {
    addedProperty?: string;
  }

  export interface ExplicitlyExportedInterface {
    otherProperty?: string;
  }
}
```

## Actual result

```text
eslint(no-unused-vars): Interface 'ImplicitlyExportedInterface' is declared but never used.
```

## Expected result

No error, matching `@typescript-eslint/no-unused-vars`.

Declarations inside a declaration module are exported implicitly and do not need local references.
Oxlint only reports the interface without an explicit `export`, even though `export` is redundant in this context.
